### DISCUSSION TITLE
How to order by  included table's JSONB field

### DISCUSSION THREAD

**Discussion by sennmac** [sequelize/sequelize#14677] [Help & Questions]
hi ~dear devs~

I have some tables a like
budget
`{id:''}`
budgetDetail 
`{id:'',budgetIdl:'',detail:{amount:1}}`

I need to search on the budget table and include the budgetDetail table then order by the detail->amount.

so I write code like 
`Budget.findAll({include:[budgetDetail],order:[Models.budgetDetail,Models.sequelize.literal("detail->>'amount'")]})`

but you know the amount is a text in jsonb, so I need to convert it into a number.

`Budget.findAll({include:[budgetDetail],order:[Models.budgetDetail, Models.sequelize.fn('TO_NUMBER', Models.sequelize.literal("budgetDetail".detail->>'amount'"),]})`

But the raw SQL will render into 

` ORDER BY "budgetDetail".TO_NUMBER("detail->>'amount'") ASC`

which will fail.

![image](https://user-images.githubusercontent.com/841597/175566693-a576ae7f-27d5-4769-9843-6ad79381a8eb.png)

**Comment by ephys** [comment]
You don't need to specify the model if you use a literal or fn:

```js
Budget.findAll({
  include: [BudgetDetail],
  order: [fn('TO_NUMBER', literal(`"budgetDetail".detail->>'amount'"`)],
});
```

**Comment by sennmac** [comment]
Thanks for your reply.
I just find out that I used the wrong operator.
I should use this one `->`, `->` will return the right type.
And for `->>` will always return text in jsonb.

Again, thank you.~
